Assets/Scripts/UI/DialogBox/FileSaveLoad/DialogBoxSaveGame.cs

```diff
@@ -93,9 +93,16 @@ public IEnumerator OkayWasClickedCoroutine()
             dbm.dialogBoxPromptOrInfo.SetPrompt("message_saving_game");
             dbm.dialogBoxPromptOrInfo.ShowDialog();
 
-            yield return new WaitForSecondsRealtime(.05f);
+            // Skip a frame so that user will see pop-up
+            yield return null;
 
-            SaveWorld(filePath);
+            Thread t = SaveWorld(filePath);
+
+            // Wait for data to be saved to HDD.
+            while (t.IsAlive)
+            {
+                yield return null;
+            }
 
```

Comment:
If there was an exception while saving there is something really wrong so currently exceptions go to the top of a call tree.